comp/snmpscan/impl/devicescan.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package snmpscanimpl
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+	"github.com/DataDog/datadog-agent/pkg/snmp/gosnmplib"
+	"github.com/gosnmp/gosnmp"
+)
+
+func (s snmpScannerImpl) RunDeviceScan(snmpConnection *gosnmp.GoSNMP, deviceNamespace string) error {
+	forwarder, err := s.demux.GetEventPlatformForwarder()
+	if err != nil {
+		return fmt.Errorf("unable to get sender: %w", err)
+	}
```

Comment:
We could move this code to the component initialize function and return an error the event platform is not properly initialized. That would catch issue earlier, and it would avoid having to call this function everytime we execute the device scan command.

```go
func NewComponent(reqs Requires) (Provides, error) {
	forwarder, err := s.demux.GetEventPlatformForwarder()
	if err != nil {
		return Provides{}, err
	}
	scanner := snmpScannerImpl{
		log:   reqs.Logger,
		forwarder: forwarder,
	}
	provides := Provides{
		Comp: scanner,
	}
	return provides, nil
}
```

Reply:
Great idea!